Add RBAC templates to Kommander (#3201)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.2/tutorials/configure-rbac/index.md

You have now created role bindings that associate the Global Admin group with these two roles, which ensure that the users in the Global Admin group have admin access to the management cluster and all target clusters.

#### Using kubectl

Assuming `kubectl` is attached to your Kommander cluster and you have a `VirtualGroup` named `global-admin`. You can apply these two `VirtualGroupKommanderClusterRoleBinding` to link the group to roles:

```yaml
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupKommanderClusterRoleBinding
metadata:
  name: global-admin-cluster-role-binding
spec:
  clusterRoleRef:
    name: cluster-admin
  virtualGroupRef:
    name: global-admin
---
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupKommanderClusterRoleBinding
metadata:
  name: global-admin-kommander-role-binding
spec:
  clusterRoleRef:
    name: kommander-global-admin
  virtualGroupRef:
    name: global-admin
```

### Global Viewer persona

The _Global Viewer_ persona should have read-only access to all resources on the management cluster and all target clusters.

> The two cluster roles for this persona are:

Reviewer: ```suggestion
This persona has the following roles:
```

- _View Role_: a `ClusterRole` applies to all target clusters in all workspaces.
- _Global Kommander View Role_: a `ClusterRole` applies to the management cluster.

#### Using the Kommander UI

In the Kommander UI, do the following:

1. Select _Global_ in the header drop-down.
2. Select _Administration_ > _Access Control_.

> 3. Select the _Cluster Roles_ tab. Kommander creates several default roles, the ones to select for this persona are _View Role_ and _Global Kommander View Role_.

Reviewer: ```suggestion
3. Select the _Cluster Roles_ tab. Kommander creates several default roles. Select  _View Role_ and _Global Kommander View Role_ for this persona.
```

4. Select the Cluster Policies tab and add these two roles to the _Global Viewer_ group.

> You have now created role bindings that associate the Global Viewer group with these two roles, which ensure that the users in the Global Viewer group have read-only access to the management cluster and all target clusters.

Reviewer: ```suggestion
You now have role bindings that associate the Global Viewer group with these two roles. The roles ensure that the users in the Global Viewer group have read-only access to the management cluster and all target clusters.
```

#### Using kubectl

> Assuming `kubectl` is attached to your Kommander cluster and you have a `VirtualGroup` named `global-viewer`. You can apply these two `VirtualGroupKommanderClusterRoleBinding` to link the group to roles:

Reviewer: ```suggestion
Assuming you have `kubectl` on your Kommander cluster and a `VirtualGroup` named `global-viewer`, you can apply these two `VirtualGroupKommanderClusterRoleBinding` to link the group to roles:
```

```yaml
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupKommanderClusterRoleBinding
metadata:
  name: workspace-admin-cluster-role-binding
spec:
  clusterRoleRef:
    name: view
  virtualGroupRef:
    name: workspace-admin
---
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupKommanderClusterRoleBinding
metadata:
  name: workspace-admin-kommander-role-binding
spec:
  clusterRoleRef:
    name: kommander-global-view
  virtualGroupRef:
    name: workspace-admin
```

### Workspace Admin persona

> The _Workspace Admin_ persona should have admin access to the Default Workspace. This allows users to administrate all namespaced resources in the workspace namespace on the management cluster, as well as all cluster resources on the workspace’s target clusters.

Reviewer: ```suggestion
The _Workspace Admin_ persona should have admin access to the Default Workspace. This allows users to administer all namespaced resources in the workspace namespace on the management cluster and all cluster resources on the workspace’s target clusters.
```

> The two roles for this persona are:

Reviewer: ```suggestion
This persona has the following roles:
```

- _Workspace Admin Role_: a `WorkspaceRole` that applies to all target clusters in all workspaces

> - _Kommander Workspace Admin Role_: a `KommanderWorkspaceRole` that applies to the management cluster. This role grants admin access to the workspace namespace on the management cluster. When a `VirtualGroupKommanderWorkspaceRoleBinding` with this role gets created, a `ClusterRole` and `VirtualGroupKommanderClusterRoleBinding` will automatically be created at the global level to allow access to this workspace specifically, instead of all workspaces.

Reviewer: ```suggestion
- _Kommander Workspace Admin Role_: a `KommanderWorkspaceRole` that applies to the management cluster. This role grants admin access to the workspace namespace on the management cluster. When a `VirtualGroupKommanderWorkspaceRoleBinding` with this role gets created, a `ClusterRole` and `VirtualGroupKommanderClusterRoleBinding` automatically create at the global level to allow access to this specific workspace, instead of all workspaces.
```

#### Using the Kommander UI

In the Kommander UI, do the following:

1. Select _Global_ in the header drop-down.
2. Select _Administration_ > _Access Control_.

> 3. Select the _Cluster Roles_ tab. The two roles for this persona are _Workspace Admin Role_ and _Kommander Workspace Admin Role_.

Reviewer: ```suggestion
3. Select the _Cluster Roles_ tab. This persona has these roles: _Workspace Admin Role_ and _Kommander Workspace Admin Role_.
```

4. Select the Cluster Policies tab and assign these two roles to the _Workspace Admin_ group.

#### Using kubectl

> Assuming `kubectl` is attached to your Kommander cluster and you have a `VirtualGroup` named `workspace-admin`. You can apply this template to link the group to roles:

Reviewer: ```suggestion
Assuming you have `kubectl` on your Kommander cluster and a `VirtualGroup` named `workspace-admin`, you can apply this template to link the group to roles:
```

```yaml
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupWorkspaceRoleBinding
metadata:
  name: workspace-admin-role-binding
  namespace: <namespace-of-workspacerole-workspace-admin>
spec:
  workspaceRoleRef:
    name: workspace-admin
  virtualGroupRef:
    name: workspace-admin
---
apiVersion: workspaces.kommander.mesosphere.io/v1alpha1
kind: VirtualGroupKommanderWorkspaceRoleBinding
metadata:
  name: workspace-admin-kommander-role-binding
  namespace: <namespace-of-kommanderworkspacerole-kommander-workspace-admin>
spec:
  kommanderWorkspaceRoleRef:
    name: kommander-workspace-admin
  virtualGroupRef:
    name: workspace-admin
```

### Workspace Viewer persona

> The _Workspace Viewer_ persona will grant users in a speific group read access to the Default Workspace. Users in the group can view all namespaced resources in the workspace namespace on the management cluster, as well as all cluster resources on the workspace’s target clusters.

Reviewer: ```suggestion
The _Workspace Viewer_ persona grants users in a specific group read access to the Default Workspace. Users in the group can view all namespaced resources in the workspace namespace on the management cluster and all cluster resources on the workspace’s target clusters.
```

> The two roles for this persona are:

Reviewer: ```suggestion
This persona has the following roles:
```

- _Workspace View Role_: a `WorkspaceRole` that applies to all target clusters in all workspaces

> - _Kommander Workspace View Role_: a `KommanderWorkspaceRole` that applies to the management cluster. This role grants read access to the workspace namespace on the management cluster. When a `VirtualGroupKommanderWorkspaceRoleBinding` with this role gets created, a `ClusterRole` and `VirtualGroupKommanderClusterRoleBinding` will automatically be created at the global level to allow access to this workspace specifically, instead of all workspaces.

Reviewer: ```suggestion
- _Kommander Workspace View Role_: a `KommanderWorkspaceRole` that applies to the management cluster. This role grants read access to the workspace namespace on the management cluster. When a `VirtualGroupKommanderWorkspaceRoleBinding` with this role gets created, a `ClusterRole` and `VirtualGroupKommanderClusterRoleBinding` create at the global level to allow access to this specific workspace, instead of all workspaces.
```

#### Using the Kommander UI

In the Kommander UI, do the following:

1. Select _Global_ in the header drop-down.
2. Select _Administration_ > _Access Control_.

> 3. Select the _Cluster Roles_ tab. The two roles for this persona are _Workspace View Role_ and _Kommander Workspace View Role_.

Reviewer: ```suggestion
3. Select the _Cluster Roles_ tab. This persona has the following roles:  _Workspace View Role_ and _Kommander Workspace View Role_.
```

4. Select the Cluster Policies tab and assign these two roles to the _Workspace Viewer_ group.

#### Using kubectl

> Assuming `kubectl` is attached to your Kommander cluster and you have a `VirtualGroup` named `workspace-viewer`. You can apply this template to link the group to roles:

Reviewer: ```suggestion
Assuming you have `kubectl` on your Kommander cluster and have a `VirtualGroup` named `workspace-viewer`, you can apply this template to link the group to roles:
```
